11-payment-encoding.md

```diff
@@ -293,6 +293,7 @@ on the amount sent.
 
 A writer:
   - if `payment_secret` feature is set:
+    - MUST set the `var_onion_optin` feature.
```

Comment:
This means that our two latest test vectors must be updated as they are not spec-compliant as they set the `payment_secret` feature bit but not the `var_onion_optin` one.

See:

* On mainnet, please send $30 for coffee beans to the same peer, which supports features 15 and 99
* On mainnet, please send $30 for coffee beans to the same peer, which supports features 15, 99 and 100

They should also set feature 9.

Reply:
good catch @t-bast! i'll update the test vectors as well

Reply:
@cdecker @rustyrussell will CL invoices set `var_onion_optin` in the next release?

Reply:
It looks like they do, @niftynei pointed to that part of the code: https://github.com/ElementsProject/lightning/blob/master/common/features.c#L60
So we should be good to go once we've got another ACK

Reply:
@TheBlueMatt pointed out in https://github.com/lightningnetwork/lightning-rfc/pull/723 that the spec currently doesn't allow `var_onion_optin` in a Bolt 11 features bitfield, which was an oversight.
Since you're adding a line to explicitly mention that it MUST be set in the invoice, we should update Bolt 9 to say `IN9`.

Reply:
@t-bast @TheBlueMatt added `9` to `var_onion_optin` context

Reply:
ah i see it's also done in #723, happy to rebase if it goes in first

Reply:
@cfromknecht looks like you added that `9` on `gossip_queries` instead of `var_onion_optin` ;)